paddle/framework/scope.h

```diff
@@ -56,7 +56,9 @@ class Scope {
     if (var) {
       return var;
     } else {
-      vars_[name] = std::unique_ptr<Variable>(new Variable());
+      auto ptr = new Variable();
+      vars_[name] = std::unique_ptr<Variable>(ptr);
+      var_names_[ptr] = name;
```

Comment:
Because in Python API, the user may want to use `Variable*` to make operator. 

```python
out = Network.fc(x="x", w="w")
out2 = Network.fc(x=out, w="w2")  # here, out is a Variable*, not std::string
```